### PR TITLE
Added support for custom c flags in nxbuild.bat

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: src/nxbuild.bat [hunk 1]
```diff
@@ -121,6 +121,7 @@ goto :BUILD
 @set TARGETLIB=libluajit_%TARGETLIB_SUFFIX%.a
 :BUILD
 del %TARGETLIB%
+@set LJCOMPILE=%LJCOMPILE% %TARGET_CFLAGS%
```
Comment (JCash) [inline src/nxbuild.bat:124]: I needed this to specify `-fPIC`

**Comment by MikePall** [comment]
Umm, the script is building a `*.a` library, which must be statically linked. Why does this need `-fPIC`?
Or are you turning this into a shared library? Or integrating it into a shared library? Are shared libraries with dynamic relocation a supported concept on the NX?

**Comment by JCash** [comment]
Very good question.
Unfortunately I do not have a good answer for you. This is part of the recommended build settings 🤷

**Comment by MikePall** [comment]
Well, if it's recommended ... now `-fPIC` is always on.

**Pull request closed without merging**